test/libsolidity/SemanticTest.cpp

```diff
@@ -52,6 +52,28 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 	m_lineOffset(m_reader.lineNumber()),
 	m_enforceViaYul(enforceViaYul)
 {
+	auto simpleSmokeBuiltin = [](FunctionCall const& _call) -> std::optional<bytes> {
+		// This function is only used in test/libsolidity/semanticTests/builtins/smoke.sol.
+		// It could be removed when we have actual builtins.
+		std::optional<bytes> result;
+		if (_call.arguments.parameters.size() < 3)
+		{
+			result = bytes();
+			for (auto const& parameter: _call.arguments.parameters)
+				result.value() += util::toBigEndian(u256{util::fromHex(parameter.rawString)});
+		}
+		return result;
+	};
+	m_builtins["smoke_test0"] = simpleSmokeBuiltin;
+	m_builtins["smoke_test1"] = simpleSmokeBuiltin;
+	m_builtins["smoke_test2"] = simpleSmokeBuiltin;
+
+	m_hooks.emplace_back([](FunctionCall const& _call) -> std::vector<std::string> {
+	  if (_call.signature.find("smoke_") != string::npos)
+			return {_call.signature};
+		return {};
+	});
```

Comment:
Could you please add another hook? Wondering what a test with multiple hooks looks like and perhaps it will also result in effects with more than one string element in the vector.